File: Algorithms/179_Largest_Number/Solution.scala
object Solution {
  def largestNumber(nums: Array[Int]): String = nums.map(_.toString)
    .sortWith((a, b) => a + b > b + a)
    .mkString("")
    .dropWhile(_ == '0') match {
    case "" => "0"
    case s => s
  }
}

File: Algorithms/179_Largest_Number/Solution.kt
class Solution {
    fun largestNumber(nums: IntArray): String = nums.map { it.toString() }
            .sortedWith(Comparator { a, b -> (b + a).compareTo(a + b) })
            .joinToString("")
            .let {
                0.until(it.length - 1)
                        .takeWhile { i -> it[i] == '0' }
                        .lastOrNull()
                        ?.let { i -> it.substring(i + 1, it.length) }
                        ?: it
            }
}